<?xml version="1.0" encoding="utf-8"?>
<xliff xmlns="urn:oasis:names:tc:xliff:document:1.2" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="1.2" xsi:schemaLocation="urn:oasis:names:tc:xliff:document:1.2 xliff-core-1.2-transitional.xsd">
  <file datatype="xml" original="NullReferenceException.xml" source-language="en-US" target-language="it-IT">
    <header>
      <tool tool-id="mdxliff" tool-name="mdxliff" tool-version="1.0-15c36f0" tool-company="Microsoft" />
      <xliffext:skl_file_name xmlns:xliffext="urn:microsoft:content:schema:xliffextensions">02cd5861-7ce2-4a82-b358-31f8435a0ac5871adebd50877699047b62f9d350cf2555dfe480.skl</xliffext:skl_file_name>
      <xliffext:version xmlns:xliffext="urn:microsoft:content:schema:xliffextensions">1.2</xliffext:version>
      <xliffext:ms.openlocfilehash xmlns:xliffext="urn:microsoft:content:schema:xliffextensions">871adebd50877699047b62f9d350cf2555dfe480</xliffext:ms.openlocfilehash>
      <xliffext:ms.sourcegitcommit xmlns:xliffext="urn:microsoft:content:schema:xliffextensions">d31dc2ede16f6f7bc64e90d9f897ff54c4e3869b</xliffext:ms.sourcegitcommit>
      <xliffext:ms.lasthandoff xmlns:xliffext="urn:microsoft:content:schema:xliffextensions">04/03/2018</xliffext:ms.lasthandoff>
      <xliffext:moniker_ids xmlns:xliffext="urn:microsoft:content:schema:xliffextensions">netcore-1.0,netcore-1.1,netcore-2.0,netcore-2.1,netframework-4.5.1,netframework-4.5.2,netframework-4.5,netframework-4.6.1,netframework-4.6.2,netframework-4.6,netframework-4.7.1,netframework-4.7,netstandard-1.0,netstandard-1.1,netstandard-1.2,netstandard-1.3,netstandard-1.4,netstandard-1.5,netstandard-1.6,netstandard-2.0,xamarinandroid-7.1,xamarinios-10.8,xamarinmac-3.0</xliffext:moniker_ids>
    </header>
    <body>
      <group id="content" extype="content">
        <trans-unit id="101" translate="yes" xml:space="preserve" uid="T:System.NullReferenceException">
          <source>The exception that is thrown when there is an attempt to dereference a null object reference.</source>
          <target state="translated">Eccezione generata quando viene effettuato un tentativo di dereferenziare un oggetto Null.</target>       </trans-unit>
        <trans-unit id="102" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>A <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> exception is thrown when you try to access a member on a type whose value is <ph id="ph2">`null`</ph>.</source>
          <target state="translated">Un'eccezione <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> viene generata quando si tenta di accedere a un membro in un tipo il cui valore è <ph id="ph2">`null`</ph>.</target>       </trans-unit>
        <trans-unit id="103" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>A <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> exception typically reflects developer error and is thrown in the following scenarios:</source>
          <target state="translated">Oggetto <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> eccezione in genere riflette errore developer e viene generata nei seguenti scenari:</target>       </trans-unit>
        <trans-unit id="104" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>You've forgotten to instantiate a reference type.</source>
          <target state="translated">È stata dimenticata creare un'istanza di un tipo di riferimento.</target>       </trans-unit>
        <trans-unit id="105" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>In the following example, <ph id="ph1">`names`</ph> is declared but never instantiated:</source>
          <target state="translated">Nell'esempio seguente, <ph id="ph1">`names`</ph> viene dichiarato ma mai creata un'istanza:</target>       </trans-unit>
        <trans-unit id="106" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>Some compilers issue a warning when they compile this code.</source>
          <target state="translated">Alcuni compilatori durante la compilazione di tale codice generano un avviso.</target>       </trans-unit>
        <trans-unit id="107" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>Others issue an error, and the compilation fails.</source>
          <target state="translated">Altri generano un errore e la compilazione ha esito negativo.</target>       </trans-unit>
        <trans-unit id="108" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>To address this problem, instantiate the object so that its value is no longer <ph id="ph1">`null`</ph>.</source>
          <target state="translated">Per risolvere questo problema, creare un'istanza dell'oggetto in modo che il valore non sia più <ph id="ph1">`null`</ph>.</target>       </trans-unit>
        <trans-unit id="109" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>The following example does this by calling a type's class constructor.</source>
          <target state="translated">Nell'esempio seguente viene eseguita questa operazione chiamando il costruttore della classe di un tipo.</target>       </trans-unit>
        <trans-unit id="110" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>You've forgotten to dimension an array before initializing it.</source>
          <target state="translated">È stata dimenticata alla dimensione di matrice prima della relativa inizializzazione.</target>       </trans-unit>
        <trans-unit id="111" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>In the following example, <ph id="ph1">`values`</ph> is declared to be an integer array, but the number of elements that it contains is never specified.</source>
          <target state="translated">Nell'esempio seguente, <ph id="ph1">`values`</ph> è dichiarato come una matrice di interi, ma non è stato specificato il numero di elementi in esso contenuti.</target>       </trans-unit>
        <trans-unit id="112" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>The attempt to initialize its values therefore thrown a  <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> exception.</source>
          <target state="translated">Il tentativo di inizializzare i relativi valori pertanto generati un <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> eccezione.</target>       </trans-unit>
        <trans-unit id="113" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>You can eliminate the exception by declaring the number of elements in the array before initializing it, as the following example does.</source>
          <target state="translated">È possibile eliminare l'eccezione dichiarando il numero di elementi nella matrice prima dell'inizializzazione, come nell'esempio seguente.</target>       </trans-unit>
        <trans-unit id="114" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>For more information on declaring and initializing arrays, see <bpt id="p1">[</bpt>Arrays<ept id="p1">](http://msdn.microsoft.com/library/bb79bdde-e570-4c30-adb0-1dd5759ae041)</ept> and <bpt id="p2">[</bpt>Arrays<ept id="p2">](http://msdn.microsoft.com/library/dbf29737-b589-4443-bee6-a27588d9c67e)</ept>.</source>
          <target state="translated">Per ulteriori informazioni sulla dichiarazione e inizializzazione di matrici, vedere <bpt id="p1">[</bpt>matrici<ept id="p1">](http://msdn.microsoft.com/library/bb79bdde-e570-4c30-adb0-1dd5759ae041)</ept> e <bpt id="p2">[</bpt>matrici<ept id="p2">](http://msdn.microsoft.com/library/dbf29737-b589-4443-bee6-a27588d9c67e)</ept>.</target>       </trans-unit>
        <trans-unit id="115" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>You get a <bpt id="p1">**</bpt>null<ept id="p1">**</ept> return value from a method, and then call a method on the returned type.</source>
          <target state="translated">Ottenere un <bpt id="p1">**</bpt>null<ept id="p1">**</ept> valore restituito da un metodo e quindi chiamare un metodo per il tipo restituito.</target>       </trans-unit>
        <trans-unit id="116" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>This sometimes is the result of a documentation error; the documentation fails to note that a method call can return <ph id="ph1">`null`</ph>.</source>
          <target state="translated">Questo è a volte il risultato di un errore di documentazione. la documentazione non riesce a tenere presente che una chiamata al metodo può restituire <ph id="ph1">`null`</ph>.</target>       </trans-unit>
        <trans-unit id="117" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>In other cases, your code erroneously assumes that the method will always return a non-<bpt id="p1">**</bpt>null<ept id="p1">**</ept> value.</source>
          <target state="translated">In altri casi, il codice suppone erroneamente che il metodo restituirà sempre un non -<bpt id="p1">**</bpt>null<ept id="p1">**</ept> valore.</target>       </trans-unit>
        <trans-unit id="118" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>The code in the following example assumes that the <ph id="ph1">&lt;xref:System.Array.Find%2A?displayProperty=nameWithType&gt;</ph> method always returns <ph id="ph2">`Person`</ph> object whose <ph id="ph3">`FirstName`</ph> field matches a search string.</source>
          <target state="translated">Il codice nell'esempio seguente si presuppone che il <ph id="ph1">&lt;xref:System.Array.Find%2A?displayProperty=nameWithType&gt;</ph> metodo restituisce sempre <ph id="ph2">`Person`</ph> i cui <ph id="ph3">`FirstName`</ph> campo corrisponde a una stringa di ricerca.</target>       </trans-unit>
        <trans-unit id="119" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>Because there is no match, the runtime throws a <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> exception.</source>
          <target state="translated">Poiché non esiste alcuna corrispondenza, il runtime genera un <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> eccezione.</target>       </trans-unit>
        <trans-unit id="120" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>To address this problem, test the method's return value to ensure that it is not <ph id="ph1">`null`</ph> before calling any of its members, as the following example does.</source>
          <target state="translated">Per risolvere questo problema, verificare il valore restituito del metodo per assicurarsi che non è <ph id="ph1">`null`</ph> prima di chiamare i relativi membri, come nell'esempio seguente.</target>       </trans-unit>
        <trans-unit id="121" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>You're using an expression (for example, you're chaining a list of methods or properties together) to retrieve a value and, although you're checking whether the value is <ph id="ph1">`null`</ph>, the runtime still throws a <ph id="ph2">&lt;xref:System.NullReferenceException&gt;</ph> exception.</source>
          <target state="translated">Si utilizza un'espressione (ad esempio, si concatena un elenco di metodi o proprietà insieme) per recuperare un valore e, sebbene si stia controllando se il valore è <ph id="ph1">`null`</ph>, il runtime genera comunque un <ph id="ph2">&lt;xref:System.NullReferenceException&gt;</ph> eccezione.</target>       </trans-unit>
        <trans-unit id="122" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>This occurs because one of the intermediate values in the expression returns <ph id="ph1">`null`</ph>.</source>
          <target state="translated">Questo errore si verifica perché uno dei valori intermedi nell'espressione restituisce <ph id="ph1">`null`</ph>.</target>       </trans-unit>
        <trans-unit id="123" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>As a result, your rest for <ph id="ph1">`null`</ph> is never evaluated.</source>
          <target state="translated">Di conseguenza, il resto per <ph id="ph1">`null`</ph> non viene mai valutata.</target>       </trans-unit>
        <trans-unit id="124" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>The following example defines a <ph id="ph1">`Pages`</ph> object that  caches information about web pages, which are presented by <ph id="ph2">`Page`</ph> objects.</source>
          <target state="translated">L'esempio seguente definisce un <ph id="ph1">`Pages`</ph> oggetto che memorizza nella cache di informazioni sulle pagine web, che vengono presentati da <ph id="ph2">`Page`</ph> oggetti.</target>       </trans-unit>
        <trans-unit id="125" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>The <ph id="ph1">`Example.Main`</ph> method checks whether the current web page has a non-null title and, if it does, displays the title.</source>
          <target state="translated">Il <ph id="ph1">`Example.Main`</ph> metodo controlla se la pagina web corrente ha un titolo diverso da null e, in caso affermativo, viene visualizzato il titolo.</target>       </trans-unit>
        <trans-unit id="126" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>Despite this check, however, the method throws a <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> exception.</source>
          <target state="translated">Nonostante questa verifica, tuttavia, il metodo genera un <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> eccezione.</target>       </trans-unit>
        <trans-unit id="127" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>The exception is thrown because <ph id="ph1">`pages.CurrentPage`</ph> returns <ph id="ph2">`null`</ph> if no page information is  stored in the cache.</source>
          <target state="translated">L'eccezione viene generata perché <ph id="ph1">`pages.CurrentPage`</ph> restituisce <ph id="ph2">`null`</ph> se nessuna informazione di pagina è memorizzata nella cache.</target>       </trans-unit>
        <trans-unit id="128" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>This exception can be corrected by testing the value of the <ph id="ph1">`CurrentPage`</ph> property before retrieving the current <ph id="ph2">`Page`</ph> object's <ph id="ph3">`Title`</ph> property, as the following example does:</source>
          <target state="translated">Questa eccezione può essere corretti verificando il valore di <ph id="ph1">`CurrentPage`</ph> proprietà prima del recupero corrente <ph id="ph2">`Page`</ph> dell'oggetto <ph id="ph3">`Title`</ph> proprietà, nell'esempio seguente effettua:</target>       </trans-unit>
        <trans-unit id="129" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>You're enumerating the elements of an array that contains reference types, and your attempt to process one of the elements throws a <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> exception.</source>
          <target state="translated">Si sta enumerando gli elementi di una matrice che contiene i tipi di riferimento e il tentativo di processo di uno degli elementi genera un <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> eccezione.</target>       </trans-unit>
        <trans-unit id="130" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>The following example defines a string array.</source>
          <target state="translated">L'esempio seguente definisce una matrice di stringhe.</target>       </trans-unit>
        <trans-unit id="131" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>A <ph id="ph1">`for`</ph> statement enumerates the elements in the array and calls each string's <ph id="ph2">&lt;xref:System.String.Trim%2A&gt;</ph> method before displaying the string.</source>
          <target state="translated">Oggetto <ph id="ph1">`for`</ph> istruzione enumera gli elementi nella matrice e chiama ogni stringa <ph id="ph2">&lt;xref:System.String.Trim%2A&gt;</ph> metodo prima di visualizzare la stringa.</target>       </trans-unit>
        <trans-unit id="132" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>This exception occurs if you assume that each element of the array must contain a non-null value, and the value of the array element is in fact <ph id="ph1">`null`</ph>.</source>
          <target state="translated">Questa eccezione si verifica se si presuppone che ogni elemento della matrice deve contenere un valore non null e il valore dell'elemento della matrice è in realtà <ph id="ph1">`null`</ph>.</target>       </trans-unit>
        <trans-unit id="133" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>The exception can be eliminated by testing whether the element is <ph id="ph1">`null`</ph> before performing any operation on that element, as the following example shows.</source>
          <target state="translated">L'eccezione può essere eliminata verificando se l'elemento è <ph id="ph1">`null`</ph> prima di eseguire operazioni su tale elemento, come nell'esempio seguente mostra.</target>       </trans-unit>
        <trans-unit id="134" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>A <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> exception is thrown by a method that is passed <ph id="ph2">`null`</ph>.</source>
          <target state="translated">Oggetto <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> eccezione viene generata da un metodo che viene passato <ph id="ph2">`null`</ph>.</target>       </trans-unit>
        <trans-unit id="135" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>Some methods validate the arguments that are passed to them.</source>
          <target state="translated">Alcuni metodi convalidano gli argomenti ricevuti.</target>       </trans-unit>
        <trans-unit id="136" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>If they do and one of the arguments is <ph id="ph1">`null`</ph>, the method throws an <ph id="ph2">&lt;xref:System.ArgumentNullException?displayProperty=nameWithType&gt;</ph> exception.</source>
          <target state="translated">In tal caso e se uno degli argomenti è <ph id="ph1">`null`</ph>, il metodo genera un'eccezione <ph id="ph2">&lt;xref:System.ArgumentNullException?displayProperty=nameWithType&gt;</ph>.</target>       </trans-unit>
        <trans-unit id="137" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>Otherwise, it throws a <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> exception.</source>
          <target state="translated">In caso contrario, il metodo genera un'eccezione <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph>.</target>       </trans-unit>
        <trans-unit id="138" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>The following example illustrates this scenario.</source>
          <target state="translated">Nell'esempio seguente viene illustrato questo scenario.</target>       </trans-unit>
        <trans-unit id="139" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>To address this issue, make sure that the argument passed to the method is not <ph id="ph1">`null`</ph>, or handle the thrown exception in a <ph id="ph2">`try…catch…finally`</ph> block.</source>
          <target state="translated">Per risolvere questo problema, assicurarsi che l'argomento passato al metodo non sia <ph id="ph1">`null`</ph> oppure gestire l'eccezione generata in un blocco <ph id="ph2">`try…catch…finally`</ph>.</target>       </trans-unit>
        <trans-unit id="140" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>For more information, see <bpt id="p1">[</bpt>Exceptions<ept id="p1">](http://msdn.microsoft.com/library/f99a1d29-a2a8-47af-9707-9909f9010735)</ept>.</source>
          <target state="translated">Per ulteriori informazioni, vedere <bpt id="p1">[</bpt>eccezioni<ept id="p1">](http://msdn.microsoft.com/library/f99a1d29-a2a8-47af-9707-9909f9010735)</ept>.</target>       </trans-unit>
        <trans-unit id="141" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>The following Microsoft intermediate language (MSIL) instructions throw <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph>: <ph id="ph2">`callvirt`</ph>, <ph id="ph3">`cpblk`</ph>, <ph id="ph4">`cpobj`</ph>, <ph id="ph5">`initblk`</ph>, <ph id="ph6">`ldelem.&lt;type&gt;`</ph>, <ph id="ph7">`ldelema`</ph>, <ph id="ph8">`ldfld`</ph>, <ph id="ph9">`ldflda`</ph>, <ph id="ph10">`ldind.&lt;type&gt;`</ph>, <ph id="ph11">`ldlen`</ph>, <ph id="ph12">`stelem.&lt;type&gt;`</ph>, <ph id="ph13">`stfld`</ph>, <ph id="ph14">`stind.&lt;type&gt;`</ph>, <ph id="ph15">`throw`</ph>, and <ph id="ph16">`unbox`</ph>.</source>
          <target state="translated">Le seguenti istruzioni MSIL (Microsoft Intermediate Language) generano <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph>: <ph id="ph2">`callvirt`</ph>, <ph id="ph3">`cpblk`</ph>, <ph id="ph4">`cpobj`</ph>, <ph id="ph5">`initblk`</ph>, <ph id="ph6">`ldelem.&lt;type&gt;`</ph>, <ph id="ph7">`ldelema`</ph>, <ph id="ph8">`ldfld`</ph>, <ph id="ph9">`ldflda`</ph>, <ph id="ph10">`ldind.&lt;type&gt;`</ph>, <ph id="ph11">`ldlen`</ph>, <ph id="ph12">`stelem.&lt;type&gt;`</ph>, <ph id="ph13">`stfld`</ph>, <ph id="ph14">`stind.&lt;type&gt;`</ph>, <ph id="ph15">`throw`</ph> e <ph id="ph16">`unbox`</ph>.</target>       </trans-unit>
        <trans-unit id="142" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source><ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> uses the HRESULT COR_E_NULLREFERENCE, which has the value 0x80004003.</source>
          <target state="translated"><ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> utilizza HRESULT COR_E_NULLREFERENCE, con valore 0x80004003.</target>       </trans-unit>
        <trans-unit id="143" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>For a list of initial property values for an instance of <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph>, see the <ph id="ph2">&lt;xref:System.NullReferenceException.%23ctor%2A&gt;</ph> constructors.</source>
          <target state="translated">Per un elenco di valori di proprietà iniziali per un'istanza di <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph>, vedere il <ph id="ph2">&lt;xref:System.NullReferenceException.%23ctor%2A&gt;</ph> costruttori.</target>       </trans-unit>
        <trans-unit id="144" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source><bpt id="p1">**</bpt>Handling NullReferenceException in release code<ept id="p1">**</ept></source>
          <target state="translated"><bpt id="p1">**</bpt>Gestione NullReferenceException nel codice di rilascio<ept id="p1">**</ept></target>       </trans-unit>
        <trans-unit id="145" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>It's usually better to avoid a NullReferenceException than to handle it after it occurs.</source>
          <target state="translated">È in genere preferibile evitare un'eccezione NullReferenceException rispetto per gestirla dopo si verifica.</target>       </trans-unit>
        <trans-unit id="146" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>Handling an exception can make your code harder to maintain and understand, and can sometimes introduce other bugs.</source>
          <target state="translated">La gestione di un'eccezione può complicare la gestione e la comprensione del codice e può talvolta introdurre altri bug.</target>       </trans-unit>
        <trans-unit id="147" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>A NullReferenceException is often a non-recoverable error.</source>
          <target state="translated">Un'eccezione NullReferenceException è spesso un errore irreversibile.</target>       </trans-unit>
        <trans-unit id="148" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>In these cases, letting the exception stop the app might be the best alternative.</source>
          <target state="translated">In questi casi, lasciare che l'eccezione arresti l'app potrebbe essere l'alternativa migliore.</target>       </trans-unit>
        <trans-unit id="149" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>However, there are many situations where handling the error can be useful:</source>
          <target state="translated">In diverse situazioni, tuttavia, la gestione dell'errore può essere utile:</target>       </trans-unit>
        <trans-unit id="150" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>Your app can ignore objects that are null.</source>
          <target state="translated">L'app può ignorare gli oggetti null.</target>       </trans-unit>
        <trans-unit id="151" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>For example, if your app retrieves and processes records in a database, you might be able to ignore some number of bad records that result in null objects.</source>
          <target state="translated">Ad esempio, se l'app recupera ed elabora i record di un database, potrebbe essere possibile ignorare alcuni record che risultano in oggetti null.</target>       </trans-unit>
        <trans-unit id="152" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>Recording the bad data in a log file or in the application UI might be all you have to do.</source>
          <target state="translated">Potrebbe essere sufficiente registrare i dati errati in un file di log o nell'interfaccia utente dell'applicazione.</target>       </trans-unit>
        <trans-unit id="153" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>You can recover from the exception.</source>
          <target state="translated">È possibile ripristinare lo stato precedente l'eccezione.</target>       </trans-unit>
        <trans-unit id="154" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>For example, a call to a web service that returns a reference type might return null if the connection is lost or the connection times out. You can attempt to reestablish the connection and try the call again.</source>
          <target state="translated">Ad esempio, una chiamata a un servizio Web che restituisce un tipo riferimento potrebbe restituire null in caso di perdita o timeout della connessione. È possibile tentare di ristabilire la connessione e ripetere la chiamata.</target>       </trans-unit>
        <trans-unit id="155" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>You can restore the state of your app to a valid state.</source>
          <target state="translated">È possibile ripristinare uno stato valido dell'app.</target>       </trans-unit>
        <trans-unit id="156" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>For example, you might be performing a multi-step task that requires you to save information to a data store before you call a method that throws a NullReferenceException.</source>
          <target state="translated">Supponiamo ad esempio di dover eseguire un'attività in più passaggi che richiede di salvare le informazioni in un archivio dati prima di chiamare un metodo che genera un'eccezione NullReferenceException.</target>       </trans-unit>
        <trans-unit id="157" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>If the uninitialized object would corrupt the data record, you can remove the previous data before you close the app.</source>
          <target state="translated">Se l'oggetto non inizializzato dovesse danneggiare il record di dati, è possibile rimuovere i dati precedenti prima di chiudere l'app.</target>       </trans-unit>
        <trans-unit id="158" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>You want to report the exception.</source>
          <target state="translated">È consigliabile segnalare l'eccezione.</target>       </trans-unit>
        <trans-unit id="159" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>For example, if the error was caused by a mistake from the user of your app, you can generate a message to help him supply the correct information.</source>
          <target state="translated">Ad esempio, se l'errore è stato causato da un errore dell'utente dell'app, è possibile generare un messaggio che lo aiuti a fornire le informazioni corrette.</target>       </trans-unit>
        <trans-unit id="160" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>You can also log information about the error to help you fix the problem.</source>
          <target state="translated">È anche possibile registrare le informazioni sull'errore per agevolare la risoluzione del problema.</target>       </trans-unit>
        <trans-unit id="161" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>Some frameworks, like ASP.NET, have a high-level exception handler that captures all errors to that the app never crashes; in that case, logging the exception might be the only way you can know that it occurs.</source>
          <target state="translated">Alcuni framework, come ASP.NET, hanno un gestore di eccezioni di alto livello che acquisisce tutti gli errori in modo tale che non si verifichino mai arresti anomali dell'app. In tal caso, la registrazione dell'eccezione potrebbe essere l'unico modo per sapere se si verifica effettivamente.</target>       </trans-unit>
        <trans-unit id="162" translate="yes" xml:space="preserve" uid="T:System.NullReferenceException">
          <source>Initializes a new instance of the <ph id="ph1">&lt;see cref="T:System.NullReferenceException" /&gt;</ph> class.</source>
          <target state="translated">Inizializza una nuova istanza della classe <ph id="ph1">&lt;see cref="T:System.NullReferenceException" /&gt;</ph>.</target>       </trans-unit>
        <trans-unit id="163" translate="yes" xml:space="preserve" uid="M:System.NullReferenceException.#ctor">
          <source>Initializes a new instance of the <ph id="ph1">&lt;see cref="T:System.NullReferenceException" /&gt;</ph> class, setting the <ph id="ph2">&lt;see cref="P:System.Exception.Message" /&gt;</ph> property of the new instance to a system-supplied message that describes the error, such as "The value 'null' was found where an instance of an object was required."</source>
          <target state="translated">Inizializza una nuova istanza della classe <ph id="ph1">&lt;see cref="T:System.NullReferenceException" /&gt;</ph>, impostando la proprietà <ph id="ph2">&lt;see cref="P:System.Exception.Message" /&gt;</ph> della nuova istanza su un messaggio fornito dal sistema in cui viene descritto l'errore, indicando ad esempio che è stato utilizzato un valore null dove era prevista un'istanza di un oggetto.</target>       </trans-unit>
        <trans-unit id="164" translate="yes" xml:space="preserve" uid="M:System.NullReferenceException.#ctor">
          <source>This message takes into account the current system culture.</source>
          <target state="translated">Questo messaggio prende in considerazione le impostazioni cultura correnti del sistema.</target>       </trans-unit>
        <trans-unit id="165" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor">
          <source>The following table shows the initial property values for an instance of <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph>.</source>
          <target state="translated">Nella tabella seguente vengono illustrati i valori di proprietà iniziali per un'istanza di <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph>.</target>       </trans-unit>
        <trans-unit id="166" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor">
          <source>Property</source>
          <target state="translated">Proprietà</target>       </trans-unit>
        <trans-unit id="167" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor">
          <source>Value</source>
          <target state="translated">Valore</target>       </trans-unit>
        <trans-unit id="168" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor">
          <source><ph id="ph1">`null`</ph>.</source>
          <target state="translated"><ph id="ph1">`null`</ph>.</target>       </trans-unit>
        <trans-unit id="169" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor">
          <source>The localized error message string.</source>
          <target state="translated">Stringa del messaggio di errore localizzato.</target>       </trans-unit>
        <trans-unit id="170" translate="yes" xml:space="preserve" uid="M:System.NullReferenceException.#ctor(System.String)">
          <source>A <ph id="ph1">&lt;see cref="T:System.String" /&gt;</ph> that describes the error.</source>
          <target state="translated"><ph id="ph1">&lt;see cref="T:System.String" /&gt;</ph> che descrive l'errore.</target>       </trans-unit>
        <trans-unit id="171" translate="yes" xml:space="preserve" uid="M:System.NullReferenceException.#ctor(System.String)">
          <source>The content of <bpt id="p1">&lt;c&gt;</bpt>message<ept id="p1">&lt;/c&gt;</ept> is intended to be understood by humans.</source>
          <target state="translated">Il contenuto del <bpt id="p1">&lt;c&gt;</bpt>messaggio<ept id="p1">&lt;/c&gt;</ept> deve essere facilmente comprensibile.</target>       </trans-unit>
        <trans-unit id="172" translate="yes" xml:space="preserve" uid="M:System.NullReferenceException.#ctor(System.String)">
          <source>The caller of this constructor is required to ensure that this string has been localized for the current system culture.</source>
          <target state="translated">Il chiamante del costruttore deve garantire che la stringa sia stata localizzata in base alle impostazioni cultura correnti del sistema.</target>       </trans-unit>
        <trans-unit id="173" translate="yes" xml:space="preserve" uid="M:System.NullReferenceException.#ctor(System.String)">
          <source>Initializes a new instance of the <ph id="ph1">&lt;see cref="T:System.NullReferenceException" /&gt;</ph> class with a specified error message.</source>
          <target state="translated">Inizializza una nuova istanza della classe <ph id="ph1">&lt;see cref="T:System.NullReferenceException" /&gt;</ph> con un messaggio di errore specificato.</target>       </trans-unit>
        <trans-unit id="174" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor(System.String)">
          <source>The following table shows the initial property values for an instance of <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph>.</source>
          <target state="translated">Nella tabella seguente vengono illustrati i valori di proprietà iniziali per un'istanza di <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph>.</target>       </trans-unit>
        <trans-unit id="175" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor(System.String)">
          <source>Property</source>
          <target state="translated">Proprietà</target>       </trans-unit>
        <trans-unit id="176" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor(System.String)">
          <source>Value</source>
          <target state="translated">Valore</target>       </trans-unit>
        <trans-unit id="177" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor(System.String)">
          <source><ph id="ph1">`null`</ph>.</source>
          <target state="translated"><ph id="ph1">`null`</ph>.</target>       </trans-unit>
        <trans-unit id="178" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor(System.String)">
          <source>The error message string.</source>
          <target state="translated">Stringa del messaggio di errore.</target>       </trans-unit>
        <trans-unit id="179" translate="yes" xml:space="preserve" uid="M:System.NullReferenceException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)">
          <source>The object that holds the serialized object data.</source>
          <target state="translated">Oggetto che contiene i dati oggetto serializzati.</target>       </trans-unit>
        <trans-unit id="180" translate="yes" xml:space="preserve" uid="M:System.NullReferenceException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)">
          <source>The contextual information about the source or destination.</source>
          <target state="translated">Informazioni contestuali sull'origine o sulla destinazione.</target>       </trans-unit>
        <trans-unit id="181" translate="yes" xml:space="preserve" uid="M:System.NullReferenceException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)">
          <source>Initializes a new instance of the <ph id="ph1">&lt;see cref="T:System.NullReferenceException" /&gt;</ph> class with serialized data.</source>
          <target state="translated">Inizializza una nuova istanza della classe <ph id="ph1">&lt;see cref="T:System.NullReferenceException" /&gt;</ph> con dati serializzati.</target>       </trans-unit>
        <trans-unit id="182" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)">
          <source>This constructor is called during deserialization to reconstitute the exception object transmitted over a stream.</source>
          <target state="translated">Questo costruttore viene chiamato durante la deserializzazione per ricostituire l'oggetto eccezione trasmesso su un flusso.</target>       </trans-unit>
        <trans-unit id="183" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)">
          <source>For more information, see <bpt id="p1">[</bpt>XML and SOAP Serialization<ept id="p1">](~/docs/standard/serialization/xml-and-soap-serialization.md)</ept>.</source>
          <target state="translated">Per ulteriori informazioni, vedere <bpt id="p1">[</bpt>serializzazione XML e SOAP<ept id="p1">](~/docs/standard/serialization/xml-and-soap-serialization.md)</ept>.</target>       </trans-unit>
        <trans-unit id="184" translate="yes" xml:space="preserve" uid="M:System.NullReferenceException.#ctor(System.String,System.Exception)">
          <source>The error message that explains the reason for the exception.</source>
          <target state="translated">Messaggio di errore in cui viene indicato il motivo dell'eccezione.</target>       </trans-unit>
        <trans-unit id="185" translate="yes" xml:space="preserve" uid="M:System.NullReferenceException.#ctor(System.String,System.Exception)">
          <source>The exception that is the cause of the current exception.</source>
          <target state="translated">Eccezione che ha determinato l'eccezione corrente.</target>       </trans-unit>
        <trans-unit id="186" translate="yes" xml:space="preserve" uid="M:System.NullReferenceException.#ctor(System.String,System.Exception)">
          <source>If the <bpt id="p1">&lt;c&gt;</bpt>innerException<ept id="p1">&lt;/c&gt;</ept> parameter is not <ph id="ph1">&lt;see langword="null" /&gt;</ph>, the current exception is raised in a <ph id="ph2">&lt;see langword="catch" /&gt;</ph> block that handles the inner exception.</source>
          <target state="translated">Se il parametro <bpt id="p1">&lt;c&gt;</bpt>innerException<ept id="p1">&lt;/c&gt;</ept> non è <ph id="ph1">&lt;see langword="null" /&gt;</ph>, l'eccezione corrente viene generata in un blocco <ph id="ph2">&lt;see langword="catch" /&gt;</ph> in cui viene gestita l'eccezione interna.</target>       </trans-unit>
        <trans-unit id="187" translate="yes" xml:space="preserve" uid="M:System.NullReferenceException.#ctor(System.String,System.Exception)">
          <source>Initializes a new instance of the <ph id="ph1">&lt;see cref="T:System.NullReferenceException" /&gt;</ph> class with a specified error message and a reference to the inner exception that is the cause of this exception.</source>
          <target state="translated">Inizializza una nuova istanza della classe <ph id="ph1">&lt;see cref="T:System.NullReferenceException" /&gt;</ph> con un messaggio di errore specificato e un riferimento all'eccezione interna che è la causa dell'eccezione corrente.</target>       </trans-unit>
        <trans-unit id="188" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor(System.String,System.Exception)">
          <source>An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <ph id="ph1">&lt;xref:System.Exception.InnerException%2A&gt;</ph> property.</source>
          <target state="translated">Un'eccezione generata come risultato diretto di un'eccezione precedente deve includere un riferimento all'eccezione precedente nella proprietà <ph id="ph1">&lt;xref:System.Exception.InnerException%2A&gt;</ph>.</target>       </trans-unit>
        <trans-unit id="189" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor(System.String,System.Exception)">
          <source>The <ph id="ph1">&lt;xref:System.Exception.InnerException%2A&gt;</ph> property returns the same value that is passed into the constructor, or <ph id="ph2">`null`</ph> if the <ph id="ph3">&lt;xref:System.Exception.InnerException%2A&gt;</ph> property does not supply the inner exception value to the constructor.</source>
          <target state="translated">La proprietà <ph id="ph1">&lt;xref:System.Exception.InnerException%2A&gt;</ph> restituisce lo stesso valore passato al costruttore oppure <ph id="ph2">`null`</ph> nel caso in cui la proprietà <ph id="ph3">&lt;xref:System.Exception.InnerException%2A&gt;</ph> non fornisca il valore di eccezione interna al costruttore.</target>       </trans-unit>
        <trans-unit id="190" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor(System.String,System.Exception)">
          <source>The following table shows the initial property values for an instance of <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph>.</source>
          <target state="translated">Nella tabella seguente vengono illustrati i valori di proprietà iniziali per un'istanza di <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph>.</target>       </trans-unit>
        <trans-unit id="191" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor(System.String,System.Exception)">
          <source>Property</source>
          <target state="translated">Proprietà</target>       </trans-unit>
        <trans-unit id="192" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor(System.String,System.Exception)">
          <source>Value</source>
          <target state="translated">Valore</target>       </trans-unit>
        <trans-unit id="193" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor(System.String,System.Exception)">
          <source>The inner exception reference.</source>
          <target state="translated">Riferimento all'eccezione interna.</target>       </trans-unit>
        <trans-unit id="194" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor(System.String,System.Exception)">
          <source>The error message string.</source>
          <target state="translated">Stringa del messaggio di errore.</target>       </trans-unit>
      </group>
    </body>
  </file>
</xliff>